src/components/ChecLineChart.vue

```diff
@@ -0,0 +1,212 @@
+
+export default {
+  name: 'ChecLineChart',
+  components: {
+    LineWrapper,
+  },
+  props: {
+    /**
+     * The data to show on the graph. Each dataset is an object with a `label` and a `points` attribute. Provide the
+     * name of the dataset as the label, and an array of values as "points". The values must be numbers. Optionally you
+     * can provide a `format` attribute that can be used to format an individual point when it's displayed in a tooltip.
+     */
+    datasets: {
+      type: Array,
+      required: true,
+    },
+    fadeEdges: Boolean,
+    /**
+     * The height of the chart.
+     */
+    height: {
+      type: Number,
+      default: null,
+    },
+    /**
+     * Provide an array of strings as labels for x axis. The number of labels must match the maximum number of points
+     * in a provided dataset.
+     */
+    xLabels: {
+      type: Array,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      tooltipOverlayPosition: null,
+      tooltipContent: null,
+      yHeights: [],
+    };
+  },
+  computed: {
+    // The forecast that appears on the right of the graph has it's width determined by the number of data points we
+    // render - it's half of the space between two points. This is also used to offset the graph at the start
+    forecastWidth() {
+      return `${50 / this.xLabels.length}%`;
+    },
+    leftOffset() {
+      return {
+        marginLeft: `-${this.forecastWidth}`,
+      };
+    },
+    points() {
+      return this.datasets.map(({ points }) => points);
+    },
+    minimums() {
+      return this.datasets.map(({ min }) => min);
+    },
+    tooltipConfig() {
+      return {
+        content: this.tooltipContent,
+        show: true,
+        trigger: 'manual',
+      };
+    },
+  },
+  methods: {
+    forecastStyle(height) {
+      return {
+        // The height is given as coordinate from the top, so we need to inverse it. Also, we need to factor in the
+        // margin applied to the x axis (at the top of the component)
+        height: `calc(100% - ${height}px - 4rem)`,
+        width: `calc(${this.forecastWidth} + 5px)`,
+      };
+    },
+    handleDraw({ finalYPosition }) {
+      this.yHeights = finalYPosition;
+    },
+    handleTooltip(position, tooltipModel) {
+      if (!position) {
+        this.tooltipOverlayPosition = null;
+      }
+
+      if (!tooltipModel) {
+        return;
+      }
+
+      const { x, y, padding } = position;
+      const bounds = this.$el.getBoundingClientRect();
+      this.tooltipOverlayPosition = {
+        left: `calc(${x} - ${padding} - ${bounds.x}px)`,
+        top: `calc(${y} - ${padding} - ${bounds.y}px)`,
+      };
+
+      // Figure out what index is being hovered over
+      const { index } = tooltipModel.dataPoints[0];
+      const title = tooltipModel.title[0];
+
+      this.tooltipContent = `<div class="chec-chart__tooltip"><div class="chec-chart__tooltip-title">${title}</div>${
```

Comment:
This doesn't look ideal, I assume there wasn't an easy way to do this with templates etc?

Reply:
Hmmn. I had a similar situation with the variants work that I found an alternative solution for. I'll try something similar here.

Reply:
It looks like it's possible - but given the way I'm doing to tooltips right now (positioned on an invisible overlay) it might look just as hacky as what I've done already.

I'd prefer to just run with this for now - and deal with issues if they come up.